### PR TITLE
[Docs] Add Zcmp Instructions in CVA6 user guide and requirement specifications

### PR DESCRIPTION
**Pull request by arshadrohan5**
This PR updates the user guide and requirement specifications to add Zcmp extension Instructions.

### CODE REVIEW

File: docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst [hunk 1]
```diff
@@ -0,0 +1,169 @@
...
+   "CV32A60AX", "Implemented extension"
+   "CV32A60X", "Implemented extension"
+
+**Note**: Zcmp is primarily targeted at embedded class CPUs due to implementation complexity. Additionally, it is not compatible with architecture class profiles.  
+
+RVZcmp Code Size Reduction Instructions
+---------------------------------------
+
+Zcmp belongs to group of extensions called RISC-V Code Size Reduction Extension (Zc*). Zc* has become the superset of Standard C extension adding more 16-bit instructions to the ISA.
+Zcmp includes 16-bit macro instructions, PUSH/POP and double move, which resue the encoding for c.fsdsp instruction.
```
Comment (ASintzoff) [inline docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:37]: resue => reuse

File: docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst [hunk 2]
```diff
@@ -0,0 +1,169 @@
...
+                            switch(bytes) {
+                                4: asm("sw x[i], 0(addr)");
+                                8: asm("sd x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp-=stack_adj;
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```
Comment (ASintzoff) [inline docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:59]: It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.

File: docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst [hunk 3]
```diff
@@ -0,0 +1,169 @@
...
+                            switch(bytes) {
+                                4: asm("lw x[i], 0(addr)");
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp+=stack_adj;
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```
Comment (ASintzoff) [inline docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:82]: It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.

File: docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst [hunk 4]
```diff
@@ -0,0 +1,169 @@
...
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    asm("li a0, 0");
+                    sp+=stack_adj;
+                    asm("ret");
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```
Comment (ASintzoff) [inline docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:107]: It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.

File: docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst [hunk 5]
```diff
@@ -0,0 +1,169 @@
...
+                                4: asm("lw x[i], 0(addr)");
+                                8: asm("ld x[i], 0(addr)");
+                            }
+                            addr-=bytes;
+                        }
+                    }
+                    sp+=stack_adj;
+                    asm("ret");
+
+    **Invalid values**: reg_list not in {ra, s0-s11} and stack_adj not in {16, 32, 48, 64, 80, 96, 112, 128, 144, 160}.
```
Comment (ASintzoff) [inline docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:131]: It is difficult to know what are the invalid values for `reg_list`. According to RISC-V spec, it is not only {ra, s0-s11} which is valid but also {ra}, {ra, s0}, {ra, s0-s1}, ..., {ra, s0-s9}.

Regarding stack_adj, the invalid values are depending on XLEN.

Comment (arshadrohan5) [reply docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:131]: Will something like the following make it easy to understand?
reg_list not in [ {ra}, {ra, s0}, {ra, s0-s1}, {ra, s0-s2}, ..., {ra, s0-s8}, {ra, s0-s9}, {ra, s0-s11} ], stack_adj not in [ 16, 32, 48, 64, 80, 96, 112 ] and [ 16, 32, 48, 64, 80, 96, 112, 128, 144, 160 ] for RV32 and RV64 respectively.

Comment (ASintzoff) [reply docs/01_cva6_user/RISCV_Instructions_RVZcmp.rst:131]: > Will something like the following make it easy to understand? reg_list not in [ {ra}, {ra, s0}, {ra, s0-s1}, {ra, s0-s2}, ..., {ra, s0-s8}, {ra, s0-s9}, {ra, s0-s11} ], stack_adj not in [ 16, 32, 48, 64, 80, 96, 112 ] and [ 16, 32, 48, 64, 80, 96, 112, 128, 144, 160 ] for RV32 and RV64 respectively.

Yes, it is less ambiguous. Thanks.

**Review by jquevremont — COMMENTED** [review]
I have reviewed after the merge  :-(
The new file also needs to be referenced in docs/01_cva6_user/index.rst.

**Pull request merged**